src/components/callToActionButton/CallToActionButton.test.tsx

```diff
@@ -0,0 +1,19 @@
+import { expect, vi } from "vitest";
+import CallToActionButton from "./CallToActionButton.tsx";
+import { fireEvent, render } from "@testing-library/react";
+import { theme } from "../../theme/theme.ts";
+import { ThemeProvider } from "styled-components";
+
+test("CTA button onClick event calls function when button is clicked", () => {
+  const onClickMock = vi.fn();
```

Comment:
in general mocks are global for entire test file and can be put right after the imports